mllib/src/test/scala/org/apache/spark/mllib/stat/HypothesisTestSuite.scala

```diff
@@ -254,4 +254,115 @@ class HypothesisTestSuite extends SparkFunSuite with MLlibTestSparkContext {
     assert(rCompResult.statistic ~== rKSStat relTol 1e-4)
     assert(rCompResult.pValue ~== rKSPVal relTol 1e-4)
   }
+
```

Comment:
Are there weird edge cases here with the partitioning?  E.g. where a partition has no elements?  Or only elements from one sample?  Can we provide tests for them?